src/main/java/seedu/partyplanet/logic/commands/EditToRemoveTagCommand.java

```diff
@@ -34,45 +33,53 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getFilteredPersonList();
 
-        for (Person person: lastShownList) {
-            removeTagFromPerson(model, person);
+        for (Person person : lastShownList) {
+            checkIfHasTags(person);
         }
 
         if (editedPersons.isEmpty()) {
             return new CommandResult(MESSAGE_TAGS_NOT_REMOVED);
         }
 
+        for (Person personToEdit : editedPersons) {
+            removeTagFromPerson(model, personToEdit);
+        }
+
         model.addState();
-        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
 
         return new CommandResult(String.format(MESSAGE_REMOVE_TAGS_SUCCESS,
                 editedPersons.isEmpty() ? "" : displayPersons(editedPersons)));
     }
 
+    private void checkIfHasTags(Person personToCheck) {
```

Comment:
It seems like the method is doing more than it should, in the sense that it is checking if the person has tags, and adding it to editedPersons, which might violate some abstraction principles.

On that note, the method name is misleading, since it implies that it only checks if there are tags.

I would suggest to change this method signature to private boolean hasTags(Person personToCheck), which will return true if the person has the tags, otherwise false, and leave adding the person to editedPersons to the execute method

Reply:
Good catch! Have edited the implementation in latest commit, changed the method name to hasTags that returns a boolean and leaves the addition to execute.